### PR TITLE
Use proper CSV writing for save codes

### PR DESCRIPTION
**Pull request by heinrich5991**
Supersedes #2296.

### CODE REVIEW

File: src/engine/shared/csv.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+				io_write(File, "\"\"", 2);
+			}
+		}
+		if(Start != ColumnLength)
+		{
+			io_write(File, pColumn + Start, ColumnLength - Start);
+		}
+		io_write(File, "\"", 1);
+	}
+	io_write(File, "\n", 1);
```
Comment (def-) [inline src/engine/shared/csv.cpp:39]: \r\n on windows.

**Comment by def-** [comment]
I tried to add this to the test:
```
	const char *apCols5[] = {"\"foobar\"", "\"BAZ\""};       Expect(2, apCols5, "\"foobar\",\"BAZ\"\n");
```
But the csv module seems to add lots of quotes:
```
src/test/csv.cpp:24: Failure
Expected equality of these values:
  aBuf
    Which is: "\"\"\"foobar\"\"\",\"\"\"BAZ\"\"\"\n"
  pExpected
    Which is: "\"foobar\",\"BAZ\"\n"
```

File: src/test/csv.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	for(int i = 0; i < Read; i++)
+	{
+		EXPECT_NE(aBuf[i], 0);
+	}
+	EXPECT_STREQ(aBuf, pExpected);
+}
+
+TEST(Csv, Simple)
+{
+	const char *apCols1[] = {"a", "b"};     Expect(2, apCols1, "a,b\n");
```
Comment (def-) [inline src/test/csv.cpp:29]: The newlines in these tests will now fail on Windows, won't they?

Comment (heinrich5991) [reply src/test/csv.cpp:29]: What surprises me even more is that the tests failed to compile on Windows yet the CI shows a green status. Will fix.

**Pull request merged**